frontend_vue/src/composables/useTokens.ts

```diff
@@ -0,0 +1,96 @@
+import { TOKENS_TYPE } from '@/components/constants.ts';
+import { ref } from 'vue';
+import generateManagedToken from '@/components/tokens/my_sql/generateManagedToken';
+
+type TokenOperationType = {
+  label: string;
+  description: string;
+  documentationLink: string;
+  icon: string;
+  instruction: string;
+  getNewTokenData?: (data: any) => void;
+  getManageTokenData?: (data: any) => void;
+};
+
+type TokenOperationsType = {
+  [key: string]: TokenOperationType;
+};
+
+export function useTokens() {
+  const tokensOperations = ref<TokenOperationsType>({
+    [TOKENS_TYPE.WEBBUG]: {
+      label: 'Web Bug Token',
+      description: 'An alert when a URL is visited',
+      documentationLink: '#',
+      icon: `${TOKENS_TYPE.WEBBUG}.png`,
+      instruction: 'Copy this URL to your clipboard and use as you wish',
+      getNewTokenData: (data) => {
+        return data.token_url;
+      },
+      getManageTokenData: (data) => {
+        return data.canarydrop.generated_url;
+      },
+    },
+    [TOKENS_TYPE.DNS]: {
+      label: 'DNS Token',
+      description: 'An alert when a hostname is requested',
+      documentationLink: '#',
+      icon: `${TOKENS_TYPE.DNS}.png`,
+      instruction: 'Copy this URL to your clipboard and use as you wish',
+      getNewTokenData: (data) => {
+        return data.token_url;
+      },
+      getManageTokenData: (data) => {
+        return data.canarydrop.generated_url;
+      },
+    },
+    [TOKENS_TYPE.QRCODE]: {
+      label: 'QR Code Token',
+      description: 'Generate a QR code for physical tokens',
+      documentationLink: '#',
+      icon: `${TOKENS_TYPE.QRCODE}.png`,
+      instruction: 'Use this QR Code to token a physical location or object:',
+      getNewTokenData: (data) => {
+        return data.qrcode_png;
+      },
+      getManageTokenData: (data) => {
+        return data.qr_code;
+      },
```

Comment:
for these things... do we need them here? like can we not have these functions in the form / manage / view components? because that's where folks will look for em?